### ISSUE TITLE
Profile pictures

### ISSUE THREAD

**Issue by towerofnix** [towerofnix/decent#101]
The first thing that comes to (my) mind when you say "profile pictures" is *hosting*. Then again, does that matter? Collaterally, messages weigh a *lot* more than some 100x100 images.

Alternatively, [Libravatar](https://www.libravatar.org/) could provide a solution for having profile pictures without hosting them on the chat server. But that's not necessarily good, because it means depending on an external service, which some clients might not be capable of handling (e.g. a restricted network connection where Libravatar is blocked/not whitelisted).

So I'm biased more towards hosting ourselves, but this maybe warrants discussion.

~~Partially blocked by #77.~~

I'm assigning..

* Myself, for backend work. If we go with Libravatar, there'll need to be work to integrate it and "connect" Libravatar accounts to users. If we host the files ourselves, we'll need to have an API to update and fetch profile pictures.

* @PullJosh, to design how profile pictures should look on the client. I think you might have done some work on the codepen sketch already?

* @heyitsmeuralex, to actually implement that and put together the API and design.

**Comment by heyitsmeuralex** [comment]
>to design how profile pictures should look on the client.

PullJosh already did this, and it's implemented in the `riot` branch.

**Comment by heyitsmeuralex** [comment]
I would say we should use Libravatar as many sites use it (or Gravatar, which it fallbacks to).

>If we go with Libravatar, there'll need to be work to integrate it and "connect" Libravatar accounts to users.

The only thing we need is the email address of each user. Libravatar works by taking an md5 hash of their email address.

**Comment by towerofnix** [comment]
> ? Do you mean implement it?

Sorry, I meant for you to write the frontend stuff :package: - which it looks like you've already done.

**Comment by heyitsmeuralex** [comment]
> verifying emails

Also, that's pointless 😉 
I say we don't store users' email addresses and just md5 hash them when their email is set.
